### ISSUE TITLE
XMLHttpRequest: throw when setting "responseType" on invalid request state

### ISSUE THREAD

**Issue by kettanaito** [mswjs/interceptors#311]
If the request state is loading or done, setting the `responseType` property must throw an "Invalid state" error. This improves the interceptor's compatibility with the regular XMLHttpRequest.

**Comment by kettanaito** [comment]
Or we can explore the `Proxy` approach to XMLHttpRequest interception and stop depending on the runtime compliance altogether (like in #265). 

```js
window.XMLHttpRequest = new Proxy(window.XMLHttpRequest, {
  // Spy/modify the constructor.
  construct(this, args, nextThis) {},

  // Spy/modify method calls.
  apply(...) {}
})
```

**Comment by kettanaito** [comment]
## Released: v0.20.0 🎉

This has been released in v0.20.0!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.20.0)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.20.0)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
